lib/ContextManagerClient.js

```diff
@@ -0,0 +1,137 @@
+"use strict";
+
+var zmq = require('zeromq');
+var uuidv4 = require('uuid/v4');
+
+module.exports = class ContextManagerClient {
+  constructor(contextManagerHost, contextManagerPort, responseTimeout) {
+    this.contextMap = {};
+    this.contextResponseTimeout = responseTimeout; // time in ms
+    this.contextSocket = null;
+    this.contextManagerHost = contextManagerHost;
+    this.contextManagerPort = contextManagerPort;
+  }
+
+  init() {
+    this.contextSocket = zmq.socket('dealer');
```

Comment:
Wouldn't be safer to add a guard value here so the user would not mess everything up by calling init() again?

Reply:
The user must not use this class, but I will do it on CommandHandler.init